libs/@local/status/crate/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+//! Defines a logical status and error model that is suitable for different programming
+//! environments, including REST APIs and RPC APIs.
+
+mod status_code;
+
+use std::fmt::Debug;
+
+use serde::{Deserialize, Serialize};
+pub use status_code::StatusCode;
+
+/// The canonical shape of a response object describing the status of a request between services.
+#[derive(Clone, Eq, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Status<D>
+where
+    D: Send + Sync + Debug,
+{
+    code: StatusCode,
+    /// A developer-facing description of the status.
+    ///
+    /// Where possible, this should provide guiding advice for debugging and/or handling the error.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
```

Comment:
`default` should not be needed on `Option`
```suggestion
    #[serde(skip_serializing_if = "Option::is_none")]
```

Reply:
Is that true? We've got `default` on most of our other `Option` structs in the repo. Don't you need `default` to handle `undefined` (as opposed to `null`)?

<img width="445" alt="image" src="https://user-images.githubusercontent.com/25749103/226684068-23b328b4-8daa-4ff4-ba6a-fd726154da33.png">
<img width="546" alt="image" src="https://user-images.githubusercontent.com/25749103/226684098-93c99c5c-2f10-4229-a54b-2185b90c3609.png">
etc.

Reply:
From my most recent experience this was my observation, however, [the documentation](https://serde.rs/field-attrs.html#default) suggests something different. Let's leave it to be conservative.